haskellPackages: Assorted build fixes

rvl opened the pull request:
This is for merging into the `haskell-updates` branch (PR #339272).

Tested the build with:
```
nix build --impure --expr 'let pkgs = (import ./. { system = "x86_64-linux"; });
in pkgs.haskellPackages.ghcWithHoogle (ps: with ps; [
  gi-gtk-declarative
  gi-gtk-declarative-app-simple
  reflex-gloss
  reflex-gi-gtk
  reflex-sdl2
  pipes-extras
  pipes-http
  vivid
  dear-imgui
])'
```

Upstream PRs:
- owickstrom/gi-gtk-declarative#118
- Gabriella439/Haskell-Pipes-Extras-Library#19
- Gabriella439/Haskell-Pipes-HTTP-Library#17
- merijn/broadcast-chan#19

rvl committed:
haskellPackages.dear-imgui: Fix build

rvl committed:
haskellPackages.reflex-sdl2: not broken

rvl committed:
haskellPackages.vivid: Fix build with a small patch

Some library dependency must have stopped re-exporting 'void', so now
it needs an extra import line.

This package doesn't have any VCS repo.

rvl committed:
haskellPackages.gi-gtk-declarative: Fix build

rvl committed:
haskellPackages.reflex-gloss: jailbreak to fix the build

rvl committed:
haskellPackages.reflex-gi-gtk: jailbreak to fix the build

rvl committed:
haskellPackages.pipes-extras: jailbreak to fix the build

I'm not sure why this needs a jailbreak. There is already a hackage
revision to fix it:
https://hackage.haskell.org/package/pipes-extras-1.0.15/revisions/

ysangkok commented on pkgs/development/haskell-modules/configuration-common.nix:
```diff
@@ -1912,6 +1918,9 @@ self: super: {
   pipes-http = doJailbreak super.pipes-http;
 
+  # 2024-09-18: transformers <0.6
+  pipes-extras = doJailbreak super.pipes-extras;
```
Why is this necessary? It was fixed in revision 7: https://hackage.haskell.org/package/pipes-extras-1.0.15/revisions/

rvl replied:
I couldn't figure out why! In `hackage-packages.nix`, pipes-extras has `revision = "8"`. In any case, adding a version assertion is a good idea.

rvl replied:
Aha. The test-suite component still has `transformers <0.6`.

https://hackage.haskell.org/package/pipes-extras-1.0.15/revision/8.cabal

rvl committed:
haskellPackages.pipes-http: Drop obsolete jailbreak

rvl committed:
haskell.packages.{ghc98,ghc910}.broadcast-chan: Jailbreak

maralorn commented:
This is amazingly thorough. Thank you so much.

I was just confused by vivid missing a link until I read the commit message. Would you mind putting that message into a comment anyway? Then the next time someone tries to figure out whether they can drop the override they don’t go looking for the repo.

rvl commented:
> Would you mind putting that message into a comment anyway?

OK, done, thanks @maralorn.

Also while looking for the author's e-mail address, I noticed that [vivid-0.5.2.1](https://hackage.haskell.org/package/vivid-0.5.2.1) was released about a week ago, which fixes the compile error.

rvl committed:
haskellPackages: regenerate package set based on current config

This commit has been generated by maintainers/scripts/haskell/regenerate-hackage-packages.sh

maralorn commented:
Excellent!
